### ISSUE TITLE
Creating temporary channels

### ISSUE THREAD

**Issue by Foereaper** [mumble-voip/mumo#26]
Hello!

So I am trying to create temporary channels dynamically. Unfortunately I can't find any documentation on doing this, and neither addChannel nor setChannelState allows for creating or changing a channel to a temporary one.

Is there a way to do this with the current API?

**Comment by Foereaper** [comment]
It does not, I tried the following:

```Python
cid = server.addChannel("test", 0)
state = server.getChannelState(cid)
state.temporary = True
server.setChannelState(state)
```

If I print the state after setting temporary to true, it shows the variable is set correctly, but if I fetch the channel state after setChannelState, it is back to false. It seems as if the temporary flag is not applied in setChannelState.

**Comment by Kissaki** [comment]
That's how I would expect it to work.

Is the Make temporary permission denied via ACL in that parent channel?

**Comment by Kissaki** [comment]
I guess this doesn't make much sense.

Temporary channels disappear when the last user leaves. So you can't make a channel temporary when no user is inside it.

**Comment by Foereaper** [comment]
I haven't gotten around to creating any permissions yet, so all permissions are default. I can manually create a temporary channel with no issues.

The idea at least is to create a temporary channel and automatically move a user to said channel based on context from a client plugin. I guess I could attempt to create the channel, move the user to the channel, then set the temporary context. I'll see if this works and report back.

**Comment by Foereaper** [comment]
Yeah, that doesn't work either. Seems like setChannelState just ignores the temporary variable.

```Python
        # set the states' channel ID to the new ID and update the player, to move them to the new channel
        pstate.channel = cid
        server.setState(pstate)
        
        # now that the users' state is set, we want to see if the channels' parent is the dungeon category
        # if it is, and the channels' state is not temporary, set it as temporary
        cstate = server.getChannelState(cid)
        if(cstate.parent == self.worldChannelStore.get(-2)):
            if(cstate.temporary == False):
                cstate.temporary = True
                server.setChannelState(cstate)
```

**Comment by Kissaki** [comment]
So this would be something for the Mumble Server to implement, not Mumo.

The Mumble client does not have a "Temporary" checkbox in the edit channel dialog. It only has it in the "Add Channel" dialog. So the Mumble Client UI is consistent to the behavior we see.

Do you have a clear, viable approach how it would make sense? I guess it would be *Enable being able to change a channel to be temporary*?
If you want, you can check for and create a feature request in the Mumble repository.

I will at least add a note to the slice source.
